src/StoreMenu/index.tsx

```diff
@@ -0,0 +1,106 @@
+import { color, space, borderRadius } from '@core-ds/primitives';
+import * as React from 'react';
+import styled from 'styled-components';
+import { Menu, MenuButton, MenuList, MenuItem } from '../Menu';
+import { StoreFlag } from './StoreFlag';
+
+export interface StoreMenuProps {
+   className?: string;
+   label: string;
+   initialCountryCode?: string;
+   stores: Store[];
+   selectedCountryCode?: string;
+   onChange?: (store: Store) => void;
+}
+
+interface Store {
+   name: string;
+   countryCode: string;
+   currency: string;
+   url?: string;
+}
+
+function UnstyledStoreMenu({
+   className,
+   label,
+   initialCountryCode,
+   stores,
+   selectedCountryCode,
+   onChange,
+}: StoreMenuProps) {
+   const [internalCountryCode, setInternalCountryCode] = React.useState(initialCountryCode);
+
+   const countryCode = React.useMemo(() => {
+      return selectedCountryCode || internalCountryCode;
+   }, [selectedCountryCode, internalCountryCode]);
+
+   const handleClick = React.useCallback(
+      (store: Store) => {
+         if (countryCode !== store.countryCode) {
+            setInternalCountryCode(store.countryCode);
+            onChange?.(store);
+         }
+      },
+      [onChange]
```

Comment:
Doesn't this need countryCode as a dependency?

Reply:
I noticed react recommends using the exhaustive-deps rule as part of their eslint-plugin-react-hooks package to warn when dependencies are specified incorrectly.

Reply:
Yes `countryCode` should be a dependency. I should definitely add the eslint rule like you suggest

```suggestion
      [countryCode, onChange]
```